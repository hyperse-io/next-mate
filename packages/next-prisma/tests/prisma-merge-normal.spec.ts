import * as fs from 'node:fs';
import { promisify } from 'node:util';
import { prismaMerge } from '../src/index.js';

const readFile = promisify(fs.readFile);

describe('prisma-merge-normal', () => {
  it('merge prisma test 1', async () => {
    const luciaPrismaContent = `
    model User {
      id String @unique
      name String
    }
`;

    await prismaMerge({
      luciaPrismaModel: luciaPrismaContent,
      input: ['tests/prisma/test1.prisma'],
      output: 'tests/prisma/testMerge1.prisma',
    });

    const prismaContent = await readFile('tests/prisma/testMerge1.prisma', {
      encoding: 'utf-8',
    });
    expect(prismaContent).toMatchSnapshot();
  });

  it('merge prisma test 2', async () => {
    const luciaPrismaContent = `
    model User {
      id String @unique
      name String
    }
`;

    await prismaMerge({
      luciaPrismaModel: luciaPrismaContent,
      input: ['tests/prisma/test2.prisma'],
      output: 'tests/prisma/testMerge2.prisma',
    });

    const prismaContent = await readFile('tests/prisma/testMerge2.prisma', {
      encoding: 'utf-8',
    });
    expect(prismaContent).toMatchSnapshot();
  });

  it('merge prisma all', async () => {
    const luciaPrismaContent = `
      model User {
        id String @unique
        name String
      }
  `;

    await prismaMerge({
      luciaPrismaModel: luciaPrismaContent,
      input: ['tests/prisma/test1.prisma', 'tests/prisma/test2.prisma'],
      output: 'tests/prisma/testMerge1-2.prisma',
    });

    const prismaContent = await readFile('tests/prisma/testMerge1-2.prisma', {
      encoding: 'utf-8',
    });
    expect(prismaContent).toMatchSnapshot();
  });
});
