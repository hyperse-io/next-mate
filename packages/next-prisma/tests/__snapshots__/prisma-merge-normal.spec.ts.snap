// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prisma-merge-normal > merge prisma all 1`] = `
"// *** GENERATED BY next-prisma :: DO NOT EDIT ***
model User {
	id String @unique
	content String 
	desc String 
	name String 
}
model Post {
	id String @unique
	name String 
	createdAt DateTime @default(now())
	updatedAt DateTime @default(now()) @updatedAt
}"
`;

exports[`prisma-merge-normal > merge prisma test 1 1`] = `
"// *** GENERATED BY next-prisma :: DO NOT EDIT ***
model User {
	id String @unique
	content Int 
	desc String 
	name String 
}
model Post {
	id String @unique
	name String 
	createdAt DateTime @default(now())
	updatedAt DateTime @default(now()) @updatedAt
}"
`;

exports[`prisma-merge-normal > merge prisma test 2 1`] = `
"// *** GENERATED BY next-prisma :: DO NOT EDIT ***
model User {
	id String @unique
	name String 
	content String 
}"
`;
